### PR TITLE
session: add SQL type labels to the commit and retry logs

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
It's hard to grep the conflict error in the log files if TiDB does not distinguish the internal and the user transactions.

### What is changed and how it works?
Add the SQL type labels to the commit and retry logs

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

PTAL @lysu @tiancaiamao

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -519,17 +520,17 @@ func (s *session) retry(ctx context.Context, maxCnt uint) error {
...
 		retryCnt++
 		if retryCnt >= maxCnt {
-			log.Warnf("con:%d Retry reached max count %d", connID, retryCnt)
-			metrics.SessionRetryErrorCounter.WithLabelValues(s.getSQLLabel(), metrics.LblReachMax)
+			log.Warnf("[%s] con:%d Retry reached max count %d", label, connID, retryCnt)
+			metrics.SessionRetryErrorCounter.WithLabelValues(label, metrics.LblReachMax)
 			return errors.Trace(err)
 		}
-		log.Warnf("con:%d retryable error: %v, txn: %v", connID, err, s.txn)
+		log.Warnf("[%s] con:%d retryable error: %v, txn: %v", label, connID, err, s.txn)
```
Comment (zz-jason) [inline session/session.go]: directly print `s.txn` in the log is hard to read, can we also optimize this?

Comment (jackysp) [reply session/session.go]: Yes, it's already in our plan.

**Pull request merged**